src/Common/src/System/Runtime/InteropServices/StringBuffer.cs

```diff
@@ -0,0 +1,385 @@
+                bufferIndex: startIndex,
+                destination: this,
+                destinationIndex: _length,
+                count: count);
+        }
+
+        /// <summary>
+        /// Copy contents to the specified buffer. Destination index must be within current destination length.
+        /// Will grow the destination buffer if needed.
+        /// </summary>
+        public unsafe void CopyTo(ulong bufferIndex, StringBuffer destination, ulong destinationIndex, ulong count)
+        {
+            if (destination == null) throw new ArgumentNullException("destination");
+            if (destinationIndex > destination._length) throw new ArgumentOutOfRangeException("destinationIndex");
+            if (_length < bufferIndex + count) throw new ArgumentOutOfRangeException("count");
+
+            if (count == 0) return;
+            ulong lastIndex = destinationIndex + (ulong)count;
+            if (destination._length < lastIndex) destination.Length = lastIndex;
+
+            Buffer.MemoryCopy(
+                source: CharPointer + bufferIndex,
+                destination: destination.CharPointer + destinationIndex,
+                destinationSizeInBytes: checked((long)destination.ByteCapacity * sizeof(char)),
+                sourceBytesToCopy: checked((long)count * sizeof(char)));
+        }
+
+        /// <summary>
+        /// Copy contents from the specified string into the buffer at the given index. Start index must be within the current length of
+        /// the buffer, will grow as necessary.
+        /// </summary>
+        public unsafe void CopyFrom(ulong bufferIndex, string source, int sourceIndex = 0, int count = -1)
+        {
+            if (source == null) throw new ArgumentNullException("source");
+            if (bufferIndex > _length) throw new ArgumentOutOfRangeException("bufferIndex");
+            if (sourceIndex < 0 || sourceIndex > source.Length) throw new ArgumentOutOfRangeException("sourceIndex");
+            if (count < 0) count = source.Length - sourceIndex;
+            if (source.Length < sourceIndex + count) throw new ArgumentOutOfRangeException("count");
+
+            if (count == 0) return;
+            ulong lastIndex = bufferIndex + (ulong)count;
+            if (_length < lastIndex) Length = lastIndex;
+
+            fixed (char* content = source)
+            {
+                Buffer.MemoryCopy(
+                    source: content + sourceIndex,
+                    destination: CharPointer + bufferIndex,
+                    destinationSizeInBytes: checked((long)ByteCapacity),
+                    sourceBytesToCopy: count * sizeof(char));
+            }
+        }
+
+        /// <summary>
+        /// Split the contents into strings via the given split characters.
+        /// </summary>
+        /// <exception cref="OverflowException">Thrown if the substring is too big to fit in a string.</exception>
+        unsafe public IEnumerable<string> Split(char splitCharacter)
```

Comment:
Is this function used anywhere?  Let's remove such functions we don't actually need right now.  If we were going to add a Split method, we should do it in a more allocation-friendly way, e.g. by returning a struct-based enumerator that gives back the strings (or even just ranges if that's sufficient) on-demand, rather than needing to allocate a list, an array, all of the strings even if they won't be used by the consumer, etc. (All problems with the existing String.Split.)